mantidproject/mantid
# Engineering Diffraction GUI - run GSASIIRefineFitPeaks in a separate thread

Previously, running GSASIIRefineFitPeaks synchronously from the GSAS tab of the ED GUI locked the GUI. This PR takes care of moving the algorithm to run in a separate thread, and hence stop locking the GUI.

**To test:**

You'll need GSAS-II installed. Install it using one of the scripts in `scripts/GSAS-II` or follow the instructions [here](https://subversion.xray.aps.anl.gov/trac/pyGSAS). Test data is in [GSASIIRefineFitPeaks.zip](https://github.com/mantidproject/mantid/files/1739753/GSASIIRefineFitPeaks.zip)

Start the ED GUI by going to `Interfaces -> Diffraction -> Engineering Diffraction`. You'll be prompted to enter an RB number if you haven't opened the GUI before - do so then go to the `GSAS-II Refinement` tab.

Load the run by clicking the first **Browse** button, selecting `ENGINX_280625_focused_bank_1.nxs` from the zip, and clicking **Load**. `280625_1` should appear in the list in the bottom right of the GUI - click this and the run should appear in the plot.

Select `template_ENGINX_241391_236516_North_bank.prm` for the instrument parameter file, all three `.cif` files for the phase files, a new location to save the gsas project (`.gpx`) to, and your GSAS-II installation directory (containing `GSASIIscriptable.py`), all with their respective browse buttons. Feel free to fiddle with other controls if you feel so inclined.

Click **Run Refinement**. This time the GUI should not lock up, instead all the inputs should be disabled, and the algorithm should run in a separate thread. Make sure they are re-enabled once the algorithm has finished. Try cancelling the algorithm, closing the GUI before the algorithm has finished, and anything else you can think of that might break things.

Also code review (please!)

Fixes #21843 

**Release Notes** 
[Here](https://github.com/mantidproject/mantid/pull/21916/commits/0e513059bd39cf01bbbea4f650a66d027d0f1dc7)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
